### PR TITLE
test comments

### PR DESCRIPTION
**Pull request by FlyBooks**
test review comments

### CODE REVIEW

File: vue/musicproject/vue.config.js [hunk 1]
```diff
@@ -15,8 +15,33 @@ module.exports = {
...
+        "/",
+        "/recommend",
+        "/singer",
+        "/rank",
+        "/search",
+        "/profile",
+      ],
+      useRenderEvent: true,
+      headless: true,
+      onlyProduction: true,
```
Comment (FlyBooks) [inline vue/musicproject/vue.config.js:37]: yes, we should add it.

File: vue/musicproject/vue.config.js [hunk 2]
```diff
@@ -15,8 +15,33 @@ module.exports = {
...
+      postProcess: (route) => {
+        // Defer scripts and tell Vue it's been server rendered to trigger hydration
+        route.html = route.html
+          .replace(/<script (.*?)>/g, "<script $1 defer>")
+          .replace('id="app"', 'id="app" data-server-rendered="true"');
+        return route;
+      },
+    },
+  },
 };
```
Comment (FlyBooks) [inline vue/musicproject/vue.config.js:47]: render it in our page!

**Review by FlyBooks — COMMENTED** [review]
prerender is greater!

File: vue/musicproject/package.json [hunk 3]
```diff
@@ -33,6 +33,8 @@
     "@vue/cli-service": "~4.4.0",
     "html-loader": "^1.1.0",
     "postcss-pxtorem": "^5.1.1",
+    "prerender-spa-plugin": "^3.2.1",
+    "vue-cli-plugin-prerender-spa": "~1.1.6",
```
Comment (DUODUO25) [inline vue/musicproject/package.json:37]: reviewed

File: vue/musicproject/vue.config.js [hunk 4]
```diff
@@ -15,8 +15,33 @@ module.exports = {
...
+  pluginOptions: {
+    prerenderSpa: {
+      registry: undefined,
+      renderRoutes: [
+        "/",
+        "/recommend",
+        "/singer",
+        "/rank",
+        "/search",
+        "/profile",
```
Comment (GrKite) [inline vue/musicproject/vue.config.js:33]: kkkkk

File: vue/musicproject/vue.config.js [hunk 5]
```diff
@@ -15,8 +15,33 @@ module.exports = {
       ],
     },
   },
+
```
Comment (GrKite) [inline vue/musicproject/vue.config.js:18]: llll

File: vue/musicproject/package.json (hunk 3 above)
Comment (FlyBooks) [reply vue/musicproject/package.json:37]: hiahia, i am coming

**Pull request merged**